src/StatsdClient/Worker/AsynchronousWorker.cs

```diff
@@ -77,10 +62,10 @@ void Dequeue()
 
                     if (_handler.OnIdle())
                     {
-                        Task.Delay(waitDuration).Wait();
+                        _waiter.Wait(waitDuration);
                         waitDuration = waitDuration + waitDuration;
-                        if (waitDuration > _maxWaitDuration)
-                            waitDuration = _maxWaitDuration;
+                        if (waitDuration > MaxWaitDuration)
```

Comment:
Maybe we could do the arithmetic only when necessary to save us a couple of cycles?

```C#
if (waitDuration < MaxWaitDuration) {
  waitDuration += waitDuration;
}
```

Reply:
It would mean `waitDuration` can be `MaxWaitDuration * 2 - 1` and so greater than `MaxWaitDuration`. I prefer keeping the current version.

Reply:
Yes, that's right. I made a mistake yesterday, I guess I was tired 😅  This is indeed an exponential growth, so this is not precise.